reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClientConfig.java

```diff
@@ -867,27 +867,30 @@ static final class H2OrHttp11Codec extends ChannelInboundHandlerAdapter {
 
 		@Override
 		public void channelActive(ChannelHandlerContext ctx) {
-			SslHandler sslHandler = ctx.pipeline().get(SslHandler.class);
-			if (sslHandler == null) {
-				throw new IllegalStateException("Cannot determine negotiated application-level protocol.");
-			}
-			String protocol = sslHandler.applicationProtocol() != null ? sslHandler.applicationProtocol() : ApplicationProtocolNames.HTTP_1_1;
-			if (log.isDebugEnabled()) {
-				log.debug(format(ctx.channel(), "Negotiated application-level protocol [" + protocol + "]"));
-			}
-			if (ApplicationProtocolNames.HTTP_2.equals(protocol)) {
-				configureHttp2Pipeline(ctx.channel().pipeline(), acceptGzip, decoder, http2Settings, observer);
-			}
-			else if (ApplicationProtocolNames.HTTP_1_1.equals(protocol)) {
-				configureHttp11Pipeline(ctx.channel().pipeline(), acceptGzip, decoder, metricsRecorder, uriTagValue);
-			}
-			else {
-				throw new IllegalStateException("unknown protocol: " + protocol);
-			}
+			ChannelHandler handler = ctx.pipeline().get(NettyPipeline.SslHandler);
+			if (handler instanceof SslHandler) {
+				SslHandler sslHandler = (SslHandler) handler;
 
-			ctx.fireChannelActive();
+				String protocol = sslHandler.applicationProtocol() != null ? sslHandler.applicationProtocol() : ApplicationProtocolNames.HTTP_1_1;
+				if (log.isDebugEnabled()) {
+					log.debug(format(ctx.channel(), "Negotiated application-level protocol [" + protocol + "]"));
+				}
+				if (ApplicationProtocolNames.HTTP_2.equals(protocol)) {
+					configureHttp2Pipeline(ctx.channel().pipeline(), acceptGzip, decoder, http2Settings, observer);
+				}
+				else if (ApplicationProtocolNames.HTTP_1_1.equals(protocol)) {
+					configureHttp11Pipeline(ctx.channel().pipeline(), acceptGzip, decoder, metricsRecorder, uriTagValue);
+				}
+				else {
+					throw new IllegalStateException("unknown protocol: " + protocol);
+				}
 
-			ctx.channel().pipeline().remove(this);
+				ctx.fireChannelActive();
+
+				ctx.channel().pipeline().remove(this);
+			} else {
```

Comment:
Make checkstyle happy
`[checkstyle] [ERROR] /.../reactor-netty/reactor-netty/reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClientConfig.java:891:25: '}' at column 4 should be alone on a line. [RightCurly]`

```suggestion
			}
			else {
```